cmd/helm/search.go

```diff
@@ -60,14 +64,62 @@ func newSearchCmd(out io.Writer) *cobra.Command {
 	}
 
 	f := cmd.Flags()
-	f.BoolVarP(&o.regexp, "regexp", "r", false, "use regular expressions for searching")
-	f.BoolVarP(&o.versions, "versions", "l", false, "show the long listing, with each version of each chart on its own line")
-	f.StringVar(&o.version, "version", "", "search using semantic versioning constraints")
+	f.StringVar(&o.searchEndpoint, "endpoint", "https://hub.helm.sh", "monocular instance to query for charts")
+	f.BoolVarP(&o.repositories, "repositories", "r", false, "search repositories you have added instead of monocular")
+	f.BoolVarP(&o.regexp, "regexp", "", false, "use regular expressions for searching repositories you have added")
+	f.BoolVarP(&o.versions, "versions", "l", false, "show the long listing, with each version of each chart on its own line, for repositories you have added")
+	f.StringVar(&o.version, "version", "", "search using semantic versioning constraints on repositories you have added")
+	f.UintVar(&o.maxColWidth, "maxColumnWidth", 50, "maximum column width for output table")
 
 	return cmd
 }
 
 func (o *searchOptions) run(out io.Writer, args []string) error {
+
+	// If searching the repositories added via helm repo add follow that path
+	if o.repositories {
+		debug("searching repositories")
+		// If an endpoint was passed in but searching repositories the user should
+		// know the option is being skipped
+		if o.searchEndpoint != "https://hub.helm.sh" {
+			fmt.Fprintln(out, "Notice: Setting the \"endpoint\" flag has no effect when searching repositories you have added")
+		}
+
+		return o.runRepositories(out, args)
+	}
+
+	// Search the Helm Hub or other monocular instance
+	debug("searching monocular")
+	// If an an option used against repository searches is used the user should
+	// know the option is being skipped
+	if o.regexp {
+		fmt.Fprintln(out, "Notice: Setting the \"regexp\" flag has no effect when searching monocular (e.g., Helm Hub)")
+	}
+	if o.versions {
+		fmt.Fprintln(out, "Notice: Setting the \"versions\" flag has no effect when searching monocular (e.g., Helm Hub)")
+	}
+	if o.version != "" {
+		fmt.Fprintln(out, "Notice: Setting the \"version\" flag has no effect when searching monocular (e.g., Helm Hub)")
+	}
+
+	c, err := monocular.New(o.searchEndpoint)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("unable to create connection to %q", o.searchEndpoint))
+	}
+
+	q := strings.Join(args, " ")
```

Comment:
Do we need to perform any sanitation here? I don't think so, but want to bring it up for discussion

Reply:
Good question. The escaping happens inside the monocular package [here](https://github.com/helm/helm/blob/29d6853659e212aebe9a8e4ebcb10f4bf66989eb/internal/monocular/search.go#L108).